File: frontend/src/api/api-functions/reports.js
import { apiGet } from "../client";
import { reportEndpoints } from "../apiEndpoints";

// Expected: { quotes_created, avg_approval_hours, top_product, pipeline_value,
//             by_status: [{status,count,value}],
//             by_rep: [{rep,quotations,value,flagged_lines}],
//             filter_options: {reps: [str], categories: [str]} }
//
// filters is {days, rep, category}; any of them may be omitted.
export async function loadReports(filters = {}) {
  return apiGet(reportEndpoints.summary(filters));
}

File: backend/app/schemas/reports.py
"""Response models for the reporting screen."""

from pydantic import BaseModel

from app.schemas.auth import ErrorResponse  # noqa: F401  re-exported for routes


class StatusRow(BaseModel):
    status: str
    count: int
    value: float


class RepRow(BaseModel):
    rep: str
    quotations: int
    value: float
    flagged_lines: int


class FilterOptions(BaseModel):
    reps: list[str]
    categories: list[str]


class ReportsData(BaseModel):
    quotes_created: int
    avg_approval_hours: float | None
    top_product: str
    pipeline_value: float
    by_status: list[StatusRow]
    by_rep: list[RepRow]
    filter_options: FilterOptions


class ReportsResponse(BaseModel):
    success: bool
    message: str
    data: ReportsData
